src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -563,25 +620,36 @@ def forward(
         if self.gradient_checkpointing and self.training and use_cache:
             use_cache = False
 
-        if cache_params is None and use_cache:
-            cache_params = MambaCache(
-                self.config, inputs_embeds.size(0), device=inputs_embeds.device, dtype=inputs_embeds.dtype
-            )
+        if use_cache:
+            if cache_params is None:
+                cache_params = MambaCache(
+                    self.config, inputs_embeds.size(0), device=inputs_embeds.device, dtype=inputs_embeds.dtype
+                )
+                cache_position = torch.arange(0, self.config.conv_kernel, device=inputs_embeds.device)
```

Comment:
should just be the length of the input

Reply:
I set it to be able to know whether we are in prefill or decoding stage in a dynamo friendly way

Reply:
yes